modules/iam-users/main.tf

```diff
@@ -0,0 +1,180 @@
+        ],
+        "Resource" : "*"
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "ssm:PutParameter",
+          "ssm:GetParameter",
+          "ssm:DeleteParameter"
+        ],
+        "Resource" : "*"
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "kms:Encrypt",
+          "kms:Decrypt",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:DescribeKey"
+        ],
+        "Resource" : "*"
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        Resource = "arn:aws:logs:*:*:*"
+      }
+    ]
+  })
+}
+
+# Attach the policy to the Lambda execution role
+resource "aws_iam_role_policy_attachment" "lambda_permissions_attach" {
+  role       = aws_iam_role.lambda_exec_role.name
+  policy_arn = aws_iam_policy.lambda_permissions.arn
+}
+
+data "archive_file" "lambda_zip" {
+  type        = "zip"
+  source_file = "${path.module}/lambda/lambda.py"
+  output_path = "${path.module}/lambda/lambda.zip"
+}
+# Create a Lambda function
+resource "aws_lambda_function" "key_rotation" {
+  function_name    = var.function_name
+  filename         = data.archive_file.lambda_zip.output_path
+  handler          = "lambda.lambda_handler" # Modify this based on your specific handler configuration
+  runtime          = "python3.9"
+  timeout          = 300
+  source_code_hash = data.archive_file.lambda_zip.output_base64sha256
+  role             = aws_iam_role.lambda_exec_role.arn
+  environment {
+    variables = {
+      DYNAMODB_TABLE_NAME = aws_dynamodb_table.service_account_table.name
+    }
+  }
+}
+
+# Create a Cloud watch event rule for every 5 minutes
+resource "aws_cloudwatch_event_rule" "every_hour" {
+  name                = "every-hour"
+  schedule_expression = "rate(1 hour)"
+}
+
+# Set the Lambda function to run every 5 minutes
+resource "aws_cloudwatch_event_target" "every_hour_target" {
+  rule      = aws_cloudwatch_event_rule.every_hour.name
+  target_id = "LambdaFunction"
+  arn       = aws_lambda_function.key_rotation.arn
+}
+
+# Grant permission for the every-five-minutes event to invoke the Lambda function
+resource "aws_lambda_permission" "allow_cloudwatch_every_five_minutes" {
+  statement_id  = "AllowExecutionFromCloudWatchEveryFiveMinutes"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.key_rotation.function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.every_hour.arn
+}
+
+# Permission boundary
+resource "aws_iam_policy" "s3_full_access_boundary" {
+  name        = "BCGOV_IAM_USER_BOUNDARY_POLICY"
+  path        = "/"
+  description = "Permission boundary policy for full S3 access"
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect   = "Allow",
+        Action   = "s3:*",
```

Comment:
The permission boundary should include permission to read the `/iam_users/*` ssm parameter store values.

Reply:
This should be in a separate statement

Reply:
Yep, Right. Added that and also perms to decrypt the parameters